klock/src/commonMain/kotlin/com/soywiz/klock/PatternDateFormat.kt

```diff
@@ -126,6 +127,11 @@ data class PatternDateFormat @JvmOverloads constructor(
 	// EEE, dd MMM yyyy HH:mm:ss z -- > Sun, 06 Nov 1994 08:49:37 GMT
     // YYYY-MM-dd HH:mm:ss
 
+    val suffixes = listOf("0th",  "1st",  "2nd",  "3rd",  "4th",  "5th",  "6th",  "7th",  "8th",  "9th",
+        "10th", "11th", "12th", "13th", "14th", "15th", "16th", "17th", "18th", "19th",
+        "20th", "21st", "22nd", "23rd", "24th", "25th", "26th", "27th", "28th", "29th",
+        "30th", "31st")
```

Comment:
I think this should be part of the locale, since this is different on each language. I think as a first approach we can provide English only, then request people from different countries to help us adjusting this. Shouldn't be a big deal making this incrementally since people is not using this yet

Reply:
Additionally I believe the name could be improved. Since the list is not a list of suffixes, but an actual list of ordinals (if I remember correctly).

Reply:
When moving to the locale I see that this can be a table generated. Not sure if output side would be better with a generated table or like that, but maybe:

```
when (num % 10) {
    1 -> "st"
    2 -> "nd"
    3 -> "rd"
    else -> "th"
}
```

could be checked faster against the actual rules and should be less prone to errors, still I'm also ok with the precomputed table as long as it is part of the locale, and the precomputed tables works for all the languages (for example not depending on the month, etc.)

I'm not fully sure if the implementation is using an Appender right now, but another performance-wise option without a table could be similar this too:

```
// English
fun writeOrdinal(day: Int, out: Appender) {
       out.append(day)
       when (num) {
             ....
       }
}
```

I'll let you decide how to do this. And other people feedback in this area is also welcome since I have limited info here to provide better feedback!

Reply:
I used a precomputed table because it's easier to translate to other languages. I'm not sure if any language use different suffixes for each day. If we use a precomputed table or not I think it should be a list with 31 elements to keep the implementation simple and less prone to errors.

Reply:
What about (on the Locale class):

```
override protected val ordinals = Array(32) {
  when (it % 10) {
    1 -> "${it}st"
    2 -> "${it}nd"
    3 -> "${it}rd"
    else -> "${it}th"
  }
}
open fun getOrdinalByDay(day: Int): String = ordinals[day] // Just in case someone doesn't want to use a table
```

Then on a test (or similar):

```
assertEquals("0th 1st 2nd", locale.ordinals.joinToString(" "))
```

That way we are providing the rules on the implementation, yet we can check each value independently for errors or as a reference on the test. By being an `open val`, other languages can either provide a rule-based orginal generation, or a precomputed literal list with: `override val ordinals = arrayOf("", ...)` each language can choose what fits better for their rules.

Still I let you decide what to do here. I'm ok with literals or rules, as long as it supports localization

Reply:
Done as open val ordinals and open functions getOrdinalByDay and getDayByOrdinal.

Reply:
The code I put is wrong, since there are three irregulars: 11, 12, 13. Can you fix it?

Reply:
Done